### PR TITLE
sort out dir structure, add grpc, add clang-format

### PR DESCRIPTION
**Pull request by Veiasai**

### CODE REVIEW

**Review by Gusabary — APPROVED** [review]

**Comment by Veiasai** [comment]
https://github.com/Sarcasm/run-clang-format

File: common/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+else()
+  set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc_cpp_plugin>)
+endif()
+
+aux_source_directory(. SOURCE)
+add_subdirectory(grpc)
+add_subdirectory(home)
+add_subdirectory(tests)
+
+add_library(common ${SOURCE})
```
Comment (Veiasai) [inline common/CMakeLists.txt]: setup_library

File: common/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+target_link_libraries(common PRIVATE ftxui::component grpc++)
+
+foreach(lib common)
+  target_include_directories(${lib}
```
Comment (Veiasai) [inline common/CMakeLists.txt]: target_include_dir(${MODULE_NAME} ..)

**Review by Veiasai — COMMENTED** [review]
git action matrix.
win, linux.
debug, release.
common, valgrind.

**Pull request merged**